The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5426629525/jobs/9868923175) [(plain)](https://github.com/rust-lang/rust/commit/72398bf0acdc7ec8ed3ff730fb61bee8f31fa26c/checks/14697322209/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] tests/ui/impl-trait/in-trait/wf-bounds.rs#next stdout ----
diff of stderr:

47    |
48 LL | struct NeedsDisplay<T: Display>(T);
49    |                        ^^^^^^^ required by this bound in `NeedsDisplay`
+ help: consider restricting type parameter `T`
+    |
+ LL |     fn nya4<T: std::fmt::Display>() -> impl Wf<NeedsDisplay<T>>;
50 
51 error: aborting due to 4 previous errors
52 



The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/in-trait/wf-bounds.next/wf-bounds.next.stderr
To only update this specific test, also pass `--test-args impl-trait/in-trait/wf-bounds.rs`

error in revision `next`: 1 errors occurred comparing output.
status: exit status: 1
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/impl-trait/in-trait/wf-bounds.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--cfg" "next" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/in-trait/wf-bounds.next" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/impl-trait/in-trait/wf-bounds.next/auxiliary" "-Zlower-impl-trait-in-trait-to-assoc-ty"
stdout: none
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
##[error]  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:17:22
   |
Build completed unsuccessfully in 0:10:59
Build completed unsuccessfully in 0:10:59
LL |     fn nya() -> impl Wf<Vec<[u8]>>;
   |
   = help: the trait `Sized` is not implemented for `[u8]`
note: required by a bound in `Vec`
  --> /rustc/FAKE_PREFIX/library/alloc/src/vec/mod.rs:396:1
  --> /rustc/FAKE_PREFIX/library/alloc/src/vec/mod.rs:396:1

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
##[error]  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:20:23
   |
LL |     fn nya2() -> impl Wf<[u8]>;
   |
   = help: the trait `Sized` is not implemented for `[u8]`
note: required by a bound in `Wf`
  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:10:10
  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:10:10
   |
LL | trait Wf<T> {
   |          ^ required by this bound in `Wf`
help: consider relaxing the implicit `Sized` restriction
   |
LL | trait Wf<T: ?Sized> {

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
##[error]  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:23:44
   |
   |
LL |     fn nya3() -> impl Wf<(), Output = impl Wf<Vec<[u8]>>>;
   |
   = help: the trait `Sized` is not implemented for `[u8]`
note: required by a bound in `Vec`
  --> /rustc/FAKE_PREFIX/library/alloc/src/vec/mod.rs:396:1
  --> /rustc/FAKE_PREFIX/library/alloc/src/vec/mod.rs:396:1

error[E0277]: `T` doesn't implement `std::fmt::Display`
##[error]  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:26:26
   |
LL |     fn nya4<T>() -> impl Wf<NeedsDisplay<T>>;
   |                          ^^^^^^^^^^^^^^^^^^^ `T` cannot be formatted with the default formatter
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
note: required by a bound in `NeedsDisplay`
  --> fake-test-src-base/impl-trait/in-trait/wf-bounds.rs:14:24
   |
LL | struct NeedsDisplay<T: Display>(T);
   |                        ^^^^^^^ required by this bound in `NeedsDisplay`
help: consider restricting type parameter `T`
   |
LL |     fn nya4<T: std::fmt::Display>() -> impl Wf<NeedsDisplay<T>>;

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0277`.
```

</details>

--- a/tests/ui/impl-trait/in-trait/wf-bounds.current.stderr
+++ b/tests/ui/impl-trait/in-trait/wf-bounds.current.stderr
@@ -1,5 +1,5 @@
 error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
-  --> $DIR/wf-bounds.rs:13:22
+  --> $DIR/wf-bounds.rs:17:22
    |
 LL |     fn nya() -> impl Wf<Vec<[u8]>>;
    |                      ^^^^^^^^^^^^^ doesn't have a size known at compile-time
@@ -9,14 +9,14 @@ note: required by a bound in `Vec`
   --> $SRC_DIR/alloc/src/vec/mod.rs:LL:COL
 
 error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
-  --> $DIR/wf-bounds.rs:16:23
+  --> $DIR/wf-bounds.rs:20:23
    |
 LL |     fn nya2() -> impl Wf<[u8]>;
    |                       ^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `[u8]`
 note: required by a bound in `Wf`
-  --> $DIR/wf-bounds.rs:8:10
+  --> $DIR/wf-bounds.rs:10:10
    |
 LL | trait Wf<T> {
    |          ^ required by this bound in `Wf`
@@ -26,7 +26,7 @@ LL | trait Wf<T: ?Sized> {
    |           ++++++++
 
 error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
-  --> $DIR/wf-bounds.rs:19:44
+  --> $DIR/wf-bounds.rs:23:44
    |
 LL |     fn nya3() -> impl Wf<(), Output = impl Wf<Vec<[u8]>>>;
    |                                            ^^^^^^^^^^^^^ doesn't have a size known at compile-time
@@ -35,6 +35,23 @@ LL |     fn nya3() -> impl Wf<(), Output = impl Wf<Vec<[u8]>>>;
 note: required by a bound in `Vec`
   --> $SRC_DIR/alloc/src/vec/mod.rs:LL:COL
 
-error: aborting due to 3 previous errors
+error[E0277]: `T` doesn't implement `std::fmt::Display`
+  --> $DIR/wf-bounds.rs:26:26
+   |
+LL |     fn nya4<T>() -> impl Wf<NeedsDisplay<T>>;
+   |                          ^^^^^^^^^^^^^^^^^^^ `T` cannot be formatted with the default formatter
+   |
+   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
+note: required by a bound in `NeedsDisplay`
+  --> $DIR/wf-bounds.rs:14:24
+   |
+LL | struct NeedsDisplay<T: Display>(T);
+   |                        ^^^^^^^ required by this bound in `NeedsDisplay`
+help: consider restricting type parameter `T`
+   |
+LL |     fn nya4<T: std::fmt::Display>() -> impl Wf<NeedsDisplay<T>>;
+   |              +++++++++++++++++++
+
+error: aborting due to 4 previous errors
 
 For more information about this error, try `rustc --explain E0277`.

--- a/tests/ui/impl-trait/in-trait/wf-bounds.next.stderr
+++ b/tests/ui/impl-trait/in-trait/wf-bounds.next.stderr
@@ -1,5 +1,5 @@
 error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
-  --> $DIR/wf-bounds.rs:13:22
+  --> $DIR/wf-bounds.rs:17:22
    |
 LL |     fn nya() -> impl Wf<Vec<[u8]>>;
    |                      ^^^^^^^^^^^^^ doesn't have a size known at compile-time
@@ -9,14 +9,14 @@ note: required by a bound in `Vec`
   --> $SRC_DIR/alloc/src/vec/mod.rs:LL:COL
 
 error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
-  --> $DIR/wf-bounds.rs:16:23
+  --> $DIR/wf-bounds.rs:20:23
    |
 LL |     fn nya2() -> impl Wf<[u8]>;
    |                       ^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `[u8]`
 note: required by a bound in `Wf`
-  --> $DIR/wf-bounds.rs:8:10
+  --> $DIR/wf-bounds.rs:10:10
    |
 LL | trait Wf<T> {
    |          ^ required by this bound in `Wf`
@@ -26,7 +26,7 @@ LL | trait Wf<T: ?Sized> {
    |           ++++++++
 
 error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
-  --> $DIR/wf-bounds.rs:19:44
+  --> $DIR/wf-bounds.rs:23:44
    |
 LL |     fn nya3() -> impl Wf<(), Output = impl Wf<Vec<[u8]>>>;
    |                                            ^^^^^^^^^^^^^ doesn't have a size known at compile-time
@@ -35,6 +35,23 @@ LL |     fn nya3() -> impl Wf<(), Output = impl Wf<Vec<[u8]>>>;
 note: required by a bound in `Vec`
   --> $SRC_DIR/alloc/src/vec/mod.rs:LL:COL
 
-error: aborting due to 3 previous errors
+error[E0277]: `T` doesn't implement `std::fmt::Display`
+  --> $DIR/wf-bounds.rs:26:26
+   |
+LL |     fn nya4<T>() -> impl Wf<NeedsDisplay<T>>;
+   |                          ^^^^^^^^^^^^^^^^^^^ `T` cannot be formatted with the default formatter
+   |
+   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
+note: required by a bound in `NeedsDisplay`
+  --> $DIR/wf-bounds.rs:14:24
+   |
+LL | struct NeedsDisplay<T: Display>(T);
+   |                        ^^^^^^^ required by this bound in `NeedsDisplay`
+help: consider restricting type parameter `T`
+   |
+LL |     fn nya4<T: std::fmt::Display>() -> impl Wf<NeedsDisplay<T>>;
+   |              +++++++++++++++++++
+
+error: aborting due to 4 previous errors
 
 For more information about this error, try `rustc --explain E0277`.

--- a/tests/ui/impl-trait/in-trait/wf-bounds.rs
+++ b/tests/ui/impl-trait/in-trait/wf-bounds.rs
@@ -5,9 +5,13 @@
 #![feature(return_position_impl_trait_in_trait)]
 #![allow(incomplete_features)]
 
+use std::fmt::Display;
+
 trait Wf<T> {
     type Output;
 }
+
+struct NeedsDisplay<T: Display>(T);
 
 trait Uwu {
     fn nya() -> impl Wf<Vec<[u8]>>;
@@ -18,6 +22,9 @@ trait Uwu {
 
     fn nya3() -> impl Wf<(), Output = impl Wf<Vec<[u8]>>>;
     //~^ ERROR the size for values of type `[u8]` cannot be known at compilation time
+
+    fn nya4<T>() -> impl Wf<NeedsDisplay<T>>;
+    //~^ ERROR `T` doesn't implement `std::fmt::Display`
 }
 
 fn main() {}

